tpgtools/templates/resource.go.tmpl

```diff
@@ -235,9 +235,15 @@ func resource{{$.PathType}}Create(d *schema.ResourceData, meta interface{}) erro
 	}
 {{ end }}
 
+{{- if $.UseDCLID }}
+{{/* TODO(magic-modules-eng): When the DCL can correctly handle IDs for regional/global splits, all resources
+should be converted to use the DCL's ID method, so normalization can be uniform. */}}
+	id, err := obj.ID()
+{{- else }}
 	id, err := {{ $.IdFunction }}(d, config, "{{$.ID}}")
+{{- end }}
 	if err != nil {
-		return fmt.Errorf("Error constructing id: %s", err)
```

Comment:
Why remove this extra formatting? The message still seems appropriate to me for the `UseDCLID` case, but if we need to modify it I think that's better than fully removing. Just makes it easier to read in the logs.

Reply:
I've put the extra formatting back, lowercased in case the error isn't at the start of a sentence.